eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventlist/EventListActivity.java

```diff
@@ -80,20 +79,20 @@ protected void onCreate(Bundle savedInstanceState)
         mScreenView = new EvenListScreenViewImpl(this);
         mScreenView.init();
 
-        RetainedObjects retainedObjects = new Restoring(getLastCustomNonConfigurationInstance());
-
         mApiService = new FutureLocalServiceConnection<>(this,
                 new Intent("com.schedjoules.API").setPackage(getPackageName()));
 
-        mEventListItems = new EventListItemsImpl();
-        mListItemsProvider = retainedObjects.getOr(0,
-                new EventListItemsProviderImpl(mApiService, mEventListItems,
-                        new EventListBackgroundMessage(this), new EventListLoadingIndicatorOverlay(this)));
+        RetainedObjects retainedObjects = new Restoring(getLastCustomNonConfigurationInstance());
+        mListItemsProvider = retainedObjects.getOr(0, new EventListItemsProviderImpl(new EventListItemsImpl()));
+
+        mListItemsProvider.setApiService(mApiService);
```

Comment:
I agree that it's probably not a good idea to bundle hundreds of events, though I don't think that it's a problem on normal configuration changes. `Bundle`s are quite lazy in that regard and they only serialze/deserialize when necessary. I believe this is not happening during a normal configuration change.
We should come up with a better way to cache the results of a query through configuration changes. If we can avoid this retention method (including retained fragments) we probably can save us a lot of trouble.
For now, please move it to a (retained) Fragment, because, as said before, it should be in a Fragment anyway.
I'll check if I can find a good way to serialize `ApiQuery`s, so we can save the previous and next page queries easily.